File: vite_react/src/types/Product.ts
// src/types/Product.ts
import {Store} from "./Store.ts";

export interface Product {
    id: number;
    name: string;
    description: string;
    price: number;
    category_id: number;
    quantity: number;
    status: string;
    stores: Store[];
    // stores: Array<{ id: number; name: string }>;
}

File: fast_api/app/models/product.py
# fast_api/app/models/product.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey
from sqlalchemy.orm import relationship
from .product_store import product_store
from ..core.database import Base


class Product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, index=True)
    description = Column(String)
    price = Column(Float)
    quantity = Column(Integer, default=0)
    status = Column(String, default="в наличии")

    category_id = Column(Integer, ForeignKey("categories.id"))
    category = relationship("Category", back_populates="products")

    # Many-to-many relationship with Store
    stores = relationship('Store', secondary=product_store, back_populates='products')
